src/Clock.elm

```diff
@@ -0,0 +1,103 @@
+module Clock exposing (Model, Msg(..), init, main, subscriptions, update, view)
+
+import Browser
+import Html exposing (..)
+import Html.Attributes exposing (style)
+import Task
+import Time
+
+
+
+-- MAIN
+
+
+main =
+    Browser.element
+        { init = init
+        , view = view
+        , update = update
+        , subscriptions = subscriptions
+        }
+
+
+
+-- MODEL
+
+
+type alias Model =
+    { zone : Time.Zone
+    , time : Time.Posix
+    }
+
+
+init : () -> ( Model, Cmd Msg )
+init _ =
+    ( Model Time.utc (Time.millisToPosix 0)
+    , Task.perform AdjustTimeZone Time.here
+    )
+
+
+
+-- UPDATE
+
+
+type Msg
+    = Tick Time.Posix
+    | AdjustTimeZone Time.Zone
+
+
+update : Msg -> Model -> ( Model, Cmd Msg )
+update msg model =
+    case msg of
+        Tick newTime ->
+            ( { model | time = newTime }
+            , Cmd.none
+            )
+
+        AdjustTimeZone newZone ->
+            ( { model | zone = newZone }
+            , Cmd.none
+            )
+
+
+
+-- SUBSCRIPTIONS
+
+
+subscriptions : Model -> Sub Msg
+subscriptions model =
+    Time.every 100 Tick
```

Comment:
Instead of consuming an unnecessary `model` - use `always`.
```elm
subscriptions : Sub Msg
subscriptions =
    Time.every 100 Tick
```
and then in `main`:
```elm
Browser.element
    ...
    , subscriptions = always subscriptions
    }
```